### DISCUSSION TITLE
Configure geolocation cities level on kubernetes hosting version

### DISCUSSION THREAD

**Discussion by erik-paula-fanhero** [plausible/analytics#2754] [Self-Hosted Support]
Hello Plausible community,

I'm trying to find a way to configure geolocation regions and cities level on kubernetes instalation versioned on the hosting repository. 

Any clues of how to do that?

**Comment by ruslandoga** [comment]
@erik-paula-fanhero 👋

What have you tried so far? What were the problems?

**Reply by ruslandoga** [reply]
It should be similar to `docker-compose` setup, you'd need to mount a volume with city-level `mmdb` database (like [GeoLite2 City](https://dev.maxmind.com/geoip/geolite2-free-geolocation-data)) and point `IP_GEOLOCATION_DB` at it.

**Reply by erik-paula-fanhero** [reply]
Hi @ruslandoga , 

I'm pretty lost with this configuration because I just find information about docker-compose setup, and I got lost where I should mount the volume and inject this values. 

Is it need to be done on plausible server deployment, right? 

And about **IP_GEOLOCATION_DB**, I imagine that I should inject it as an env var with the another ones from plausible server deployment, correct?

**Reply by ruslandoga** [reply]
I'm not a k8s user myself, but I'd try to offer some guesses :)

1. Where to mount the volume?

I think this depends on your k8s installation. If it's running somewhere where you control the host, you can mount the volume to a readonly `hostPath`, this would be equivalent to the current docker compose setup where a volume is mounted with just the `mmdb` file. In the cloud k8s offering, it might be easier to put `mmdb` on s3 or similar and skip the volume setup. In both cases you first download and put `mmdb` somewhere under your control and then deploy plausible.

2. Where to put IP_GEOLOCATION_DB?

I think it can just go together with [the other env vars.](https://github.com/plausible/hosting/blob/36ada560cbd981cc9f9ef18cb4ff43cb534e5d03/kubernetes/plausible.yaml#L101-L127) It needs to point at the mounted volume or s3 url with mmdb. Like `IP_GEOLOCATION_DB=/absolute/path/to/db` or `IP_GEOLOCATION_DB=https://<bucket>.s3.amazonaws.com/...`

**Reply by erik-paula-fanhero** [reply]
Understood  @ruslandoga. 

With your tips, I believe that everything should be inserted into the `plausible.yaml` file of the instalation. 
I will try to configure it in that way and see what happens. 

Thanks you for your attention.

**Reply by erik-paula-fanhero** [reply]
Hmmm... 

All right. 
I created an S3 bucket and put my mmdb file inside it. 
I created a variable on my plausible.yaml deployment called `IP_GEOLOCATION_DB` passing the URI of my mmdb file on S3. 

Everything started fine, but now, Plausible is unable to determine the geolocation of IP. 

I'm using this plausible image:  `image: plausible/analytics:v1.5.1`

What I missing? 

P.S.: For testing purposes I give the ReadOnlyPermission for all my account buckets to EKS role to eliminate possible missing permission issues.

**Reply by erik-paula-fanhero** [reply]
This is an image of the variable injected on my pod/container.

![image](https://user-images.githubusercontent.com/119502174/225455601-5be3112e-ec63-4832-bf20-eb9b3a565954.png)

**Reply by ruslandoga** [reply]
You can try connecting to the plausible container and checking geolocation db status manually.

With Docker it'd be

```sh
docker compose exec -ti plausible bin/plausible remote
```

I guess with `kubectl` it should be something similar. `bin/plausible remote` is the command to run inside the container, it starts an interactive session (iex) with the running elixir application.

Then once inside iex:

```elixir
iex> Geolix.lookup("8.8.8.8")
# what does it output?
```

**Reply by erik-paula-fanhero** [reply]
Oh, nevermind. 

I give up to use s3 as a mounted volume and mount a persistent volume that will be updated for another service, responsible to get new version of mmdb everyweek. 

After swap to this strategy, everything works like a charm.

**Reply by erik-paula-fanhero** [reply]
Thanks for your explanation of how it works! 
It was crucial to me to find a way to solve this configuration :)
